modules/local/subsample_bam.nf

```diff
@@ -0,0 +1,43 @@
+process SUBSAMPLE_BAM {
+    tag "${meta.id}"
+    label 'process_tiny'
+
+    conda "bioconda::samtools=1.17"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.17--h00cdaf9_0' :
+        'biocontainers/samtools:1.17--h00cdaf9_0' }"
+
+    input:
+    tuple val(meta), path(mergedbam)
+
+    output:
+    tuple val(meta), path('*.bam'), emit: subsampled_bam
+    path "versions.yml",            emit: versions
+
+    shell:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    '''
+    percentage=`wc -c !{mergedbam} | cut -d$' ' -f1 | awk '{printf "%.2f\\n", 50000000000 / $0}'`
+
+    if awk "BEGIN {exit !($percentage <= 1 )}"; then
+        samtools view -s $percentage -b !{mergedbam} > !{meta.id}_subsampled.bam
```

Comment:
${percentage}

Reply:
which to change? The one in the if is being removed. The samtools parameter is referencing a shell variable, whilst {mergedbam} is a nextflow.